### DISCUSSION TITLE
Test a user flow in Next

### DISCUSSION THREAD

**Discussion by edwinboonold** [vercel/next.js#17135] [Help] [answered]
Hi all,

I'm new to Next (so far I really like it tho) and currently running into a challenge I need to overcome. In my app there is a list of options a user can choose from. When a user picks one of the options a button becomes available and the user can go to the next page where a component is loaded based on the picked option

Now I want to write a test so I can test the flow and see if when option 1 is picked component a is rendered. 

So far I can't find any documentation on how to achieve that. I and the team I work in both don't have any clue on how we could achieve this. Is there anyone that can point us in the right direction where we can build on?

I'm not looking for a complete code example but more a some pointers on how to set it up so I can learn from it aswel

Thanks in advance

**Comment by jamesmosier** [comment, marked as answer]
Hi @edwinboon. It sounds like you want to setup some end-to-end tests. [cypress](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell) is a great choice for this. Their docs should be rather universal in the sense you'd start up your Next app and cypress would navigate to it and perform actions as you specify. 

You could also look at [Puppeteer and Jest](https://blog.logrocket.com/end-to-end-testing-react-apps-with-puppeteer-and-jest-ce2f414b4fd7/) although this requires a bit more setup initially from my experience.

**Reply by edwinboonold** [reply]
Sorry for the late response but thanks I'll will look into it
